### ISSUE TITLE
If the migrations config file doesn't exist, read the database migration table.

### ISSUE THREAD

**Issue by lukearmstrong** [fuel/oil#152]
I've just started working on a project I haven't worked on before, cloned the repo down, and made the changes required and wrote a migration. When I come to run the migrations, it tries to run them all from the beginning, throwing an error.

For some reason `app/config/development/migrations.php` doesn't exist, but this is strange behaviour considering the database has a `migration` table which knows exactly what migration that database is at for each app/module/package.

Why can't the oil migrate task simply look at the database table to determine what migration it is currently on?

**Comment by lukearmstrong** [comment]
For now, I'll write a task that will read the db migration table and save it to the `app/config/<env>/migrations.php` file

**Comment by WanWizard** [comment]
I think the way migrations are processed needs to be revised, because it's very complex at the moment.

I agree that in principe the migrations table should always reflect the current state of the database, the migration folders the state of your migration code, and the config file should only be used as a control (for stuff like defining a 'current' version).

**Comment by lukearmstrong** [comment]
The project I was working on has 169 tables, all of which were either created by migrations, or created manually and then the migration wasn't run afterwards, as the migrations spanned many modules and packages - and even worse when it was pushed to the test and live servers the migrations weren't actually run, the database was just copied.

Also I discovered that the migrations config file was not in the .gitignore, but in some clone specific excludes file, which explains why the original developers never committed it, and why I never got one when cloning the project to do a change.

So there were 2 devs working on it, using the same database, sometimes running migrations, sometimes not, and neither of their migration configs were remotely complete and impossible to merge together using a diff... so I wrote this simple generic task to read the current state of the migrations run against the database and save them to a new config file.

It would have been very time consuming to deal with manually, so hopefully this helps someone.

https://gist.github.com/3932101
